ray-operator/apis/ray/v1/rayservice_types.go

```diff
@@ -20,6 +20,12 @@ const (
 	FailedToUpdateService            ServiceStatus = "FailedToUpdateService"
 )
 
+type UpgradeStrategy string
+
+const (
+	ZeroDowntime UpgradeStrategy = "zero-downtime"
```

Comment:
I suggest consider different names for the strategy that make the behavior clearer. Also use PascalCase for the string

Reply:
I named it `BlueGreenUpgrade` since its a name that has a direct link to the specific behavior, but not sure if that's enough? I could also rename it to something that describes more of the behavior such as `UpgradeThroughSwappingRayCluster`.